builder/dockerfile/evaluator.go

```diff
@@ -201,3 +201,31 @@ func (b *Builder) dispatch(stepN int, stepTotal int, ast *parser.Node) error {
 
 	return fmt.Errorf("Unknown instruction: %s", upperCasedCmd)
 }
+
+// evaluateDispatch does a simple check for syntax errors of the Dockerfile.
+// Because some of the instructions can only be validated through runtime,
+// arg, env, etc., this syntax check will not be complete and could not replace
+// the runtime check. Instead, this function is only a helper that allows
+// user to find out the obvious error in Dockerfile earlier on.
+func (b *Builder) evaluateDispatch(ast *parser.Node) error {
+	cmd := ast.Value
+	upperCasedCmd := strings.ToUpper(cmd)
+
+	// To ensure the user is given a decent error message if the platform
+	// on which the daemon is running does not support a builder command.
+	if err := platformSupports(strings.ToLower(cmd)); err != nil {
+		return err
+	}
+
+	if cmd == "onbuild" {
```

Comment:
While this is a trivial check I'd prefer if we didn't treat "ONBIULD" as special since this type of check could be done for pretty much all of the cmds, so we should be consistent and leave this function as just a check of the cmds themselves and nothing more - for now.